### DISCUSSION TITLE
Help posting image

### DISCUSSION THREAD

**Discussion by khaliddd** [sbertix/Swiftagram#120] [Q&A] [answered]
Hi, 

Endpoint to upload image to instagram is not working. 
I get success but media is `nil` and can't see any post on instagram. 

My code is : 

```swift
Endpoint.Media.Posts.upload(image: image, captioned: "test")
            .unlocking(with: secret)
            .task { (response) in
                if case .success(let unit) = response {
                    // unit.media is nil
                }
                else if case .failure(let error) = response {
                    print(error.localizedDescription)
                }
            }.resume()
```

**Comment by sbertix** [comment]
Hey @khaliddd, 
I'm unable to test it right now, I hope I can get to it later today. 
Would you be able to confirm whether it works properly WITHOUT adding a caption to the photo? Cause it might be related to Instagram marking the comment as spam and restricting the account you're using, especially if you've been testing the API a lot.

**Comment by khaliddd** [comment]
@sbertix its not working even WITHOUT caption.

**Comment by sbertix** [comment]
Hahaha "great". I hope to get to test it soon.

**Comment by sbertix** [comment, marked as answer]
Ok so I tested it extensively and the feature still works as intended, so I'm assuming it probably doesn't work your end for one of these reasons:
- the image format is unsupported (e.g. formats enabling alpha channels are invalid)
- the image colorspace is unsupported 
- the image aspect ratio is unsupported
- some other issues with the image…

But in order to be sure, do you mind changing the logging level and sending me on Telegram the output (as it for sure contains sensitive info so don't post it here)?

```swift
Endpoint.Media.Posts.upload(image: image, captioned: "test")
            .unlocking(with: secret)
            .task { (response) in
                if case .success(let unit) = response {
                    // unit.media is nil
                }
                else if case .failure(let error) = response {
                    print(error.localizedDescription)
                }
            }
            .logging(level: .full)  // Just add this.
            .resume()
```

I suggest replicating the upload test in `EndpointTests.swift` to check whether they work your end.

**Comment by khaliddd** [comment]
@sbertix it worked when I tried to upload image without alpha components in it. 
btw what should be the specific image format, colorspace and aspect ratio ? or do you have any utility function to make an image/video compatible for instagram upload ? 
thank you for your help.

**Reply by sbertix** [reply]
There's no actual specification, as Instagram tends to support new formats, or slightly change the conditions quite often. 
Keeping in mind you should have no alpha and have a reasonable aspect ratio, for everything else I just suggest trying the upload process and monitor whether it succeeds or not. 

**Swiftagram** is just a API wrapper so we don't have any image processing, and I personally don't use the endpoint so I never got too much into the technicalities of it. 
I'm sure you can find a lot of libraries on GitHub converting, let's say, `png`s to non-alpha Instagram-supported `jpg`s, available on SPM.

**Reply by khaliddd** [reply]
got it. 
Thank you
